data-table/__tests__/table.js

```diff
@@ -56,48 +64,49 @@ describe('basic tests to ensure the buttons can function well', () => {
+        // First, check the public functions
+        expect(table.getNumRows('div-id')).toEqual(numRows);
+        expect(table.getNumColumns('div-id')).toEqual(numCols);
+
+        // Note: add 1 to numRows and numCols to account for the headers
         let contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * numCols);
-        addButton[0].click();
-        contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * (numCols + 1));
+        expect(contents.length).toEqual((numRows+1) * (numCols+1));
+    }
+
+    test('check the add column button fuctionality', () => {
+        const button = document.getElementsByClassName("left-panel-button")[0];
+        assertRowsColsEquals(3, 3);
+        button.click();
+        assertRowsColsEquals(3, 4);
     });
 
     test('check the delete column button fuctionality', () => {
-        const numRows = 4;
-        const numCols = 4;
-        const addButton = document.getElementsByClassName("left-panel-button");
-        let contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * numCols);
-        addButton[1].click();
-        contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * (numCols - 1));
+        const button = document.getElementsByClassName("left-panel-button")[1];
+        assertRowsColsEquals(3, 3);
+        button.click();
+        assertRowsColsEquals(3, 2);
     });
 
     test('check the add a row button fuctionality', () => {
-        const numRows = 4;
-        const numCols = 4;
-        const addButton = document.getElementsByClassName("left-panel-button");
-        let contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * numCols);
-        addButton[2].click();
-        contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual((numRows+1) * numCols);
+        const button = document.getElementsByClassName("left-panel-button")[2];
+        assertRowsColsEquals(3, 3);
+        button.click();
+        assertRowsColsEquals(4, 3);
     });
 
     test('check the delete a row button fuctionality', () => {
-        const numRows = 4;
-        const numCols = 4;
-        const addButton = document.getElementsByClassName("left-panel-button");
-        let contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * numCols);
-        addButton[3].click();
-        contents = document.getElementsByClassName("data-table-cell");
-        expect(contents.length).toEqual(numRows * (numCols-1));
+        const button = document.getElementsByClassName("left-panel-button")[3];
+        assertRowsColsEquals(3, 3);
+        button.click();
+        assertRowsColsEquals(2, 3);
```

Comment:
Note to reviewer: the above code is changed to simplify what's happening. The functionality has not changed, but the code is now encapsulated in `assertRowColsEquals` which tests both the newly-added public functions as well as the actual cell data